src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProvider.cs

```diff
@@ -670,26 +693,88 @@ private GraphNodeId GetGraphNodeId(string projectPath, IDependencyNode nodeInfo)
             return GraphNodeId.GetNested(partialValues.ToArray());
         }
 
+        private GraphNodeId GetTopLevelGraphNodeId(string projectPath, IDependencyNode nodeInfo)
+        {
+            var partialValues = new List<GraphNodeId>();
+            partialValues.Add(GraphNodeId.GetPartial(CodeGraphNodeIdName.Assembly,
+                                                     new Uri(projectPath, UriKind.RelativeOrAbsolute)));
+            if (nodeInfo.Flags.Contains(DependencyNode.GenericDependencyFlags))
+            {
+                var projectFolder = Path.GetDirectoryName(projectPath)?.ToLowerInvariant();
+                if (nodeInfo.Flags.Contains(DependencyNode.CustomItemSpec))
+                {
+                    var name = DependencyNode.GetName(nodeInfo);
+                    if (name != null)
+                    {
+                        partialValues.Add(GraphNodeId.GetPartial(
+                            CodeGraphNodeIdName.File,
+                            new Uri(Path.Combine(projectFolder, name.ToLowerInvariant()),
+                            UriKind.RelativeOrAbsolute)));
+                    }
+                }
+                else
+                {
+                    var fullItemSpecPath = Path.GetFullPath(
+                                            Path.Combine(projectFolder,
+                                                         nodeInfo.Id.ItemSpec?.ToLowerInvariant()));
+                    if (!string.IsNullOrEmpty(fullItemSpecPath))
```

Comment:
Could this ever not be satisfied? GetFullPath will throw if the path is illegal (such as an empty string), and will otherwise return a string that cannot be null or empty.